src/sentry_plugins/github/endpoints/webhook.py

```diff
@@ -35,13 +37,67 @@ def get_external_id(username):
 
 
 class Webhook(object):
-    def __call__(self, organization, event):
+    def __call__(self, event, organization=None):
         raise NotImplementedError
 
 
+class InstallationEventWebhook(Webhook):
+    # https://developer.github.com/v3/activity/events/types/#installationevent
+    def __call__(self, event, organization=None):
+        action = event['action']
+        installation = event['installation']
+        # TODO(jess): handle uninstalls
+        if action == 'created':
+            try:
+                with transaction.atomic():
+                    Integration.objects.create(
+                        provider='github_apps',
+                        external_id=installation['id'],
+                        name=installation['account']['login'],
+                    )
+            except IntegrityError:
+                pass
+
+
+class InstallationRepositoryEventWebhook(Webhook):
+    # https://developer.github.com/v3/activity/events/types/#installationrepositoriesevent
+    def __call__(self, event, organization=None):
+        installation = event['installation']
+
+        integration = Integration.objects.get(
+            external_id=installation['id'],
+            provider='github_apps',
+        )
+
+        repos_added = event['repositories_added']
+
+        if repos_added:
+            for org_id in integration.organizations.values_list('id', flat=True):
+                for r in repos_added:
+                    config = {
+                        'name': r['full_name'],
+                    }
+                    repo, created = Repository.objects.get_or_create(
+                        organization_id=org_id,
+                        name=r['full_name'],
+                        provider='github',
+                        external_id=r['id'],
+                        defaults={
+                            'url': 'https://github.com/%s' % (r['full_name'], ),
+                            'config': config,
+                            'integration_id': integration.id,
+                        }
+                    )
+                    if not created:
+                        repo.config.update(config)
+                        repo.integration_id = integration.id
+                        repo.save()
+        # TODO(jess): what do we want to do when they're removed?
```

Comment:
def seems like something we can be reactive too

e.g. we tried querying this data, and mark an integration as 'suspended' or something in the UI (we can follow up with this later)